### ISSUE TITLE
Create ability to correct location of weed patches

### ISSUE THREAD

**Issue by nigelcharman** [EcoNet-NZ/inaturalist-to-cams#73]
The location of the weed patch is often wrong in iNaturalist. Either because the GPS location is incorrect, or because the location of the photographer is distant from the weed location.

It is important for CAMS to have the correct location so that we can find the weed when searching for it, report correctly on weeds and detect duplicate weed instances.

The iNaturalist location can only be updated by the person who created the iNaturalist observation. In some, but not all cases, we may be able to get them to update the location by sending them a message. This may also have the benefit of them checking the location in future.
 
We could just update the location in CAMS, but if the iNaturalist observation is subsequently updated the location will be reset to the iNaturalist location when it is resynchronised. In this instance we would want to keep the updated location rather than overwriting it.

However, if the location field is updated in iNaturalist, we should probably update CAMS to use the updated iNaturalist location.

A solution may be to store the iNaturalist location in a separate field in CAMS (eg "reported_location"). When resynchronising an existing observation, do not update the location if the "reported_location" is the same.

Future enhancements may be to write the updated location back to a field in iNaturalist (eg "updated_location") and/or notify the iNaturalist user that the location has been changed in CAMS and give them the option to easily update it to match.

**Comment by amazing-will** [comment]
started taking a look at this today..

**Comment by nigelcharman** [comment]
I've just checked and I did create the fields in the WeedLocations table of the iNat_to_CAMS feature layer:

Name: ReportedLocationX
Display Name: Reported Location X coord
Type: Double

Name: ReportedLocationY
Display Name: Reported Location Y coord
Type: Double

I'm wondering if these should be a Latitude/Longitude pair rather than X/Y. Let me know if you find anything?

As well as being added to the code, they'll need to be added to the config/cams_schema.json file, which is used as a one-off check at startup that the schema matches our expectations from the code.
